extensions/metadata/interface/src/main/java/org/exist/storage/md/CollectionEvents.java

```diff
@@ -29,113 +29,142 @@
-		} catch (Throwable e) {
-			MDStorageManager.LOG.fatal(e);
-		}
-	}
-
-	@Override
-	public void beforeCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI newUri) throws TriggerException {
-	}
-
-	@Override
-	public void afterCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI oldUri) throws TriggerException {
-//		System.out.println("afterCopyCollection "+collection.getURI());
-		try {
-			MDStorageManager.inst.md.copyMetas(oldUri, collection);
-		} catch (Throwable e) {
-			MDStorageManager.LOG.fatal(e);
-		}
-	}
-
-	@Override
-	public void beforeMoveCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI newUri) throws TriggerException {
-//		System.out.println("beforeMoveCollection "+collection.getURI());
-		try {
-	        for(Iterator<DocumentImpl> i = collection.iterator(broker); i.hasNext(); ) {
-	            DocumentImpl doc = i.next();
-	            MDStorageManager.inst.md.moveMetas(
-            		collection.getURI().append(doc.getFileURI()), 
-            		newUri.append(doc.getFileURI())
-        		);
-	        }
-		} catch (PermissionDeniedException e) {
-			throw new TriggerException(e);
-		}
-	}
-
-	@Override
-	public void afterMoveCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI oldUri) throws TriggerException {
-//		System.out.println("afterMoveCollection "+oldUri+" to "+collection.getURI());
-		MDStorageManager.inst.md.moveMetas(oldUri, collection.getURI());
-	}
-	
-	private void deleteCollectionRecursive(DBBroker broker, Collection collection) throws PermissionDeniedException {
-        for(Iterator<DocumentImpl> i = collection.iterator(broker); i.hasNext(); ) {
-            DocumentImpl doc = i.next();
-            MDStorageManager.inst.md.delMetas(doc.getURI());
+    @Override
+    public void configure(DBBroker broker, Collection parent, Map<String, List<? extends Object>> parameters) throws TriggerException {
+    }
+
+    @Override
+    public void beforeCreateCollection(DBBroker broker, Txn txn, XmldbURI uri) throws TriggerException {
+        // System.out.println("beforeCreateCollection "+uri);
+    }
+
+    @Override
+    public void afterCreateCollection(DBBroker broker, Txn txn, Collection collection) {
+        // System.out.println("afterCreateCollection "+collection.getURI());
+        try {
+            MDStorageManager.get().md.addMetas(collection);
+        } catch (Throwable e) {
+            MDStorageManager.LOG.fatal(e,e);
+        }
+
+//        index(collection);
+    }
+
+    @Override
+    public void beforeCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI newUri) throws TriggerException {
+    }
+
+    @Override
+    public void afterCopyCollection(DBBroker broker, Txn txn, Collection collection, XmldbURI oldUri) {
+        // System.out.println("afterCopyCollection "+collection.getURI());
+        try {
+            MDStorageManager.get().md.copyMetas(oldUri, collection);
```

Comment:
Can you please avoid direct access to member variables, it breaks encapsulation and also makes it very difficult to write mock tests for this in future. i.e. Can you please change `MDStorageManager.get().md.copyMetas(oldUri, collection)` to `MDStorageManager.get().getMetaData().copyMetas(oldUri, collection)`